### PR TITLE
doc(hansbug): add guides for unittest, visualization and code style.

### PR DESCRIPTION
**Pull request by HansBug**
Related PR in `DI-engine` repo: https://github.com/opendilab/DI-engine/pull/309

* [x] Code Style
* [x] Unittest Guide
* [x] Graphics and Visualization

### CODE REVIEW

File: source/21_code_style/index.rst [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+* For **class**, the `Camelcase Naming Convention <https://en.wikipedia.org/wiki/Camel_case>`_ which is started with capital letters shall be adopted, such as ``MyClass``. In addition, for internal classes, you can use additional underscores at the beginning, for example ``_InnerClass``.
+* All **functions** and **methods** are named with lowercase letters, numbers and underscores, such as ``my_function`` and ``my_method``.
+* For **variables**, they are all named with lowercase letters, numbers and underscores, such as ``my_var``.
+* For methods and variables belonging to a class, use a single underscore to express the protected inheritance relationship, such as ``_protected_val``. Use two underscores to express a private inheritance relationship, for example ``__private_val``.
+* For the naming of method parameters, if it is an instance method, the first parameter should be named `` self``. If it is a class rule, the first parameter should be named ``cls``. Please use ``*args`` for the variable length parameter of the list, and ``**kwargs`` for the key value parameter.
```
Comment (sailxjx) [inline source/21_code_style/index.rst:14]: ``self``

File: source/22_test/index.rst [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+Types of Unit Test
+---------------------------------
+
+In the DI-engine project, we divide the unit test into the following parts:
+
+* ``unittest`` -- functional unit test in a general sense to ensure the normal function of engineering code and the convergence of algorithm code on simple use cases.
+* ``algotest`` -- unit test for algorithm code to ensure that the algorithm code can meet the use requirements on specific use cases.
+* ``cudatest`` -- unit test for CUDA dependent features, ensure that such features function normally in the operating environment with CUDA.
+* ``envpooltest`` -- unit test for features that rely on envpool high-performance parallel computing to ensure that such features function normally.
+* ``platformtest`` -- unit test of cross platform code, ensure that the core functions of DI engine can still operate normally on MacOS and Windows platforms.
```
Comment (sailxjx) [inline source/22_test/index.rst:29]: DI-engine

File: source/22_test/index.rst [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+* ``benchmark`` -- performance test of algorithm or architecture, speed measurement is mainly carried out for relevant contents to ensure that its performance meets the requirements.
+
+
+
+How to Build Unit Test
+---------------------------------
+
+In DI engine, we use `pytest <https://docs.pytest.org/>`_ to build unit tests.
```
Comment (sailxjx) [inline source/22_test/index.rst:37]: DI-engine

File: source/22_test/index.rst [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+    :scale: 55%
+    :align: center
+
+In addition, `` pytest`` also supports assertion of thrown exceptions, as shown below
+
+.. code:: python
+
+   import pytest
+
+   def test_zero_division():
```
Comment (sailxjx) [inline source/22_test/index.rst:66]: Add @pytest.mark.unittest in the code example

File: source/22_test/index.rst [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+    :align: center
+
+
+
+Fixture and Conftest
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Fixture is a very important mechanism in ``pytest``. It can initialize the resources required for testing and pass them in as parameters of the test function for its usage. Not only that, but also the recovery of operation resources can be realized to ensure that the subsequent operation will not be affected. In addition, code reuse can be easily realized through the definition of scope.
+
+This `Chinese tutorial <https://www.cnblogs.com/linuxchao/p/linuxchao-pytest-fixture.html>`_ is written in great detail and can be used as a reference. In the existing code of DI-engine, you can refer to `ding/league/tests/test_player.py <https://github.com/opendilab/DI-engine/tree/main/ding/league/tests/test_player.py>`_.
```
Comment (sailxjx) [inline source/22_test/index.rst:83]: Find an english tutorial here, or just discard this part.

File: source/23_visual/index.rst [hunk 6]
```diff
@@ -1,4 +1,4 @@
-Graphics and Visualization
+Diagrams and Visualization
```
Comment (sailxjx) [inline source/23_visual/index.rst:1]: Can not find images in this page.

File: source/22_test/index_zh.rst [hunk 7]
```diff
@@ -0,0 +1,155 @@
+单元测试指南
+==============================
+
+单元测试的意义
+---------------------
+
+在软件工程领域，单元测试是一种测试方法，通过这种方法，对一个或多个计算机程序模块的源代码集的各个单元以及相关的控制数据、使用程序和操作程序进行测试，以确定它们是否能正确运行（摘自`维基百科 - Unit testing <https://en.wikipedia.org/wiki/Unit_testing>`_）。
```
Comment (sailxjx) [inline source/22_test/index_zh.rst:7]: Invalid url format

File: source/22_test/index_zh.rst [hunk 8]
```diff
@@ -0,0 +1,155 @@
...
+* ``platformtest``——为针对跨平台代码的单元测试，确保DI-engine的核心功能在MacOS和Windows平台上依然可以正常运行。
+* ``benchmark``——为针对算法或架构的性能测试，主要针对相关内容进行测速，确保其性能满足要求。
+
+
+
+
+如何编写单元测试
+---------------------
+
+在DI-engine中，我们使用`pytest <https://docs.pytest.org/>`_进行单元测试的搭建。
```
Comment (sailxjx) [inline source/22_test/index_zh.rst:38]: Invalid url format, please check the urls in the rest part of the docs and preview them in the browser.

File: source/22_test/index_zh.rst [hunk 9]
```diff
@@ -0,0 +1,155 @@
...
+
+对于单元测试的撰写，整体上可以参考各级代码路径下的 ``tests`` 文件夹，例如`ding/envs/env_manager/tests <https://github.com/opendilab/DI-engine/tree/main/ding/envs/env_manager/tests>`_。
+
+
+命名规范
+~~~~~~~~~~~~~~~~~~~
+
+对于单元测试，我们一般以类或函数为单位进行搭建，其命名应当满足一定的规范，具体为：
+
+* 对于函数形态的单元测试，要求函数以``test_``开头。
```
Comment (sailxjx) [inline source/22_test/index_zh.rst:48]: Keep a space between Chinese and Punctuation.

File: source/23_visual/index.rst (hunk 6 above)
Comment (HansBug) [reply source/23_visual/index.rst:1]: The images can be automatically generated due to `diagrams.mk`

File: source/22_test/index.rst (hunk 2 above)
Comment (HansBug) [reply source/22_test/index.rst]: Fixed.

File: source/22_test/index.rst (hunk 4 above)
Comment (HansBug) [reply source/22_test/index.rst:67]: Fixed.

File: source/22_test/index_zh.rst (hunk 7 above)
Comment (HansBug) [reply source/22_test/index_zh.rst]: Fixed.

File: source/22_test/index_zh.rst (hunk 8 above)
Comment (HansBug) [reply source/22_test/index_zh.rst]: Fixed.

File: source/22_test/index_zh.rst (hunk 9 above)
Comment (HansBug) [reply source/22_test/index_zh.rst]: Fixed.

File: source/22_test/index.rst (hunk 5 above)
Comment (HansBug) [reply source/22_test/index.rst]: Fixed.

**Pull request merged**